include/aspect/material_model/interface.h

```diff
@@ -537,6 +537,11 @@ namespace aspect
        */
       template <class AdditionalOutputType>
       const AdditionalOutputType *get_additional_output() const;
+
+      /**
+       * Steal the additional outputs from @p other.
+       */
+      void take_additional_outputs_from(MaterialModelOutputs<dim> &other);
```

Comment:
Since you're moving things, may I suggest naming the function `move_...` instead of `take_...`? Using standard nomenclature is useful, I think.